File: FestivalC#/Festival/Services/Service/ModelServices/IServiceShow.cs
using Festival.Model;
using System.Collections.Generic;

namespace Festival.Service
{
    public interface IServiceShow
    {
        /*
         * Saves a show 
         */
        void save(Show show);

        /*
         * Deletes a show 
         */
        void delete(int idShow);

        /*
         * Updates a show 
         */
        void update(int id, Show show);

        /*
         * Returns a show by id 
         */
        Show getById(int idShow);

        /*
         * Gets all shows 
         */
        List<Show> getAll();

        /*
         * Gets all shows for a given artist
         */
        List<Show> getShowsForArtist(int idArtist);

        /*
         * Gets all shows for a given date 
         */
        List<Show> getShowsForDate(string date);
    }
}

File: FestivalJava/Services/src/main/java/festival/ModelServices/Interfaces/IServiceShow.java
package festival.ModelServices.Interfaces;

import festival.Domain.Show;
import festival.Validation.Exceptions.ValidatorException;

import java.util.List;

/**
 * Created by Sebi on 28-Mar-17.
 */
public interface IServiceShow {
    /*
    Saves a show
     */
    void save(Show show) throws ValidatorException;

    /*
    Deletes a show
     */
    void delete(Integer idShow);

    /*
    Updates a show
     */
    void update(Integer id, Show show) throws ValidatorException;

    /*
    Gets a show by id
     */
    Show getById(Integer idShow);

    /*
    Returns a list of shows
     */
    List<Show> getAll();

    /*
    Returns all shows for a given artist
     */
    List<Show> getShowsForArtist(Integer idArtist);

    /*
    Returns all shows for a given date
     */
    List<Show> getShowsForDate(String date);
}
